src/index.js

```diff
@@ -1,35 +1,202 @@
+
+      todoArray[indx].completed = true;
+    } else {
+      todoListCheck.nextElementSibling.style.textDecoration = 'none';
+      todoListCheck.nextElementSibling.style.opacity = '1';
+
+      todoArray[indx].completed = false;
+    }
+    localStorage.setItem('todoList', JSON.stringify(todoArray));
+  });
+};
+
+document.getElementById('todo-enter').addEventListener('click', () => {
+  const descriptionTodo = InputElementTodo.value;
+  InputElementTodo.value = '';
+  localStorage.setItem('input', InputElementTodo.value);
+  todoArray.push({
+    description: descriptionTodo,
     completed: false,
-    index: 1,
-  },
-];
+    index: todoArray.length,
+  });
+  localStorage.setItem('todoList', JSON.stringify(todoArray));
+  todoListRender(descriptionTodo, false, todoArray.length - 1);
+});
 
 todoArray.forEach((todo) => {
+  const todoListItemContainer = document.createElement('div');
+  todoListItemContainer.classList.add('todo-element-padding');
+  todoListItemContainer.classList.add('todo-item-container');
+
   const todoListCheck = document.createElement('input');
   todoListCheck.setAttribute('type', 'checkbox');
   todoListCheck.setAttribute('id', `todo-list-check-${todo.index}`);
-  todoListCheck.classList.add('todo-element-padding');
-  document.getElementById('todo-list-element').appendChild(todoListCheck);
+  todoListCheck.classList.add('test-check');
+  todoListItemContainer.appendChild(todoListCheck);
 
   const todoListElement = document.createElement('span');
-  todoListElement.classList.add('todo-element-padding');
+  todoListElement.classList.add('ellipsis');
   todoListElement.innerText = todo.description;
 
-  document.getElementById('todo-list-element').appendChild(todoListElement);
+  todoListItemContainer.appendChild(todoListElement);
+
   const threeDotIcon = document.createElement('span');
   threeDotIcon.innerHTML = '&#8942;';
-  document.getElementById('todo-list-element').appendChild(threeDotIcon);
+  todoListItemContainer.appendChild(threeDotIcon);
+
+  document
+    .getElementById('todo-list-element')
+    .appendChild(todoListItemContainer);
 
   document
     .getElementById('todo-list-element')
     .appendChild(document.createElement('hr'));
+
+  todoListCheck.addEventListener('click', () => {
+    if (todoListCheck.checked === true) {
+      todoListCheck.nextElementSibling.style.textDecoration = 'line-through';
+      todoListCheck.nextElementSibling.style.opacity = '0.5';
+      const indexOfTodoElement = todoListCheck.id.split('').pop();
+      todoArray.forEach((todo) => {
+        if (todo.index === Number(indexOfTodoElement)) {
+          todo.completed = true;
+        }
+      });
+      localStorage.setItem('todoList', JSON.stringify(todoArray));
+    } else {
+      todoListCheck.nextElementSibling.style.textDecoration = 'none';
+      todoListCheck.nextElementSibling.style.opacity = '1';
+      const indexOfTodoElement = todoListCheck.id.split('').pop();
```

Comment:
- [ ] As required for the project, kindly make sure you implement a function for `Deleting` a todo_element and `Editing` a task description.

- [ ] Also, you should ensure that when deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. This is for the project requirements

- [ ] Once, you have fixed the above changes, kindly make sure all changes to the To-Do List are correctly be saved in local storage(i.e. for deleting a task item, edit a task description,...) for the project requirements

Reply:
resolved.